### ISSUE TITLE
为什么没有看到在微调的代码中有loss字段的处理

### ISSUE THREAD

**Issue by aofengdaxia** [THUDM/ChatGLM3#838]
### System Info / 系統信息

使用Lora做微调代码

### Who can help? / 谁可以帮助到您？

@Btlmd 

### Information / 问题信息

- [X] The official example scripts / 官方的示例脚本
- [ ] My own modified scripts / 我自己修改的脚本和任务

### Reproduction / 复现过程

我看到微调的[readme](https://github.com/THUDM/ChatGLM3/blob/main/finetune_demo/README.md)中有关于多轮对话微调示每种角色可以附带一个 bool 类型的 loss 字段，表示该字段所预测的内容是否参与 loss 计算。但是在微调的代码中没有看到关于loss的字段的处理。

### Expected behavior / 期待表现

请确认微调代码是否支持了loss字段呢？

**Comment by edveftps587** [comment]
![image](https://github.com/THUDM/ChatGLM3/assets/139164170/926bc4de-26c0-4e51-8ef6-ff0b72e7ad8f)
同问，在Lora微调的时候，validation loss没有打印出来，请问下是为什么呢

**Comment by aofengdaxia** [comment]
我对比了以前的commit，发现新的版本写的Lora和p-tuning的训练中忽略了这个

**Comment by zRzRzRzRzRzRzR** [comment]
继承transformer啊，不需要写

**Comment by zRzRzRzRzRzRzR** [comment]
测了一下是有验证的loss的，你数据集有验证集吗

**Comment by zRzRzRzRzRzRzR** [comment]
> ### System Info / 系統信息
> 
> 使用Lora做微调代码
> ### Who can help? / 谁可以帮助到您？
> 
> @Btlmd
> ### Information / 问题信息
> 
>     * [x]  The official example scripts / 官方的示例脚本
> 
>     * [ ]  My own modified scripts / 我自己修改的脚本和任务
> 
> 
> ### Reproduction / 复现过程
> 
> 我看到微调的[readme](https://github.com/THUDM/ChatGLM3/blob/main/finetune_demo/README.md)中有关于多轮对话微调示每种角色可以附带一个 bool 类型的 loss 字段，表示该字段所预测的内容是否参与 loss 计算。但是在微调的代码中没有看到关于loss的字段的处理。
> ### Expected behavior / 期待表现
> 
> 请确认微调代码是否支持了loss字段呢？

你数据集这么构建就行了，这是做loss mask用的
